test/Generics/protocol_typealias_cycle_3.swift

```diff
@@ -0,0 +1,9 @@
+// RUN: %target-swift-frontend -typecheck %s -requirement-machine-protocol-signatures=on
+
+protocol P {
+  typealias MyFunction =
+      @convention(c) (UnsafeMutableRawPointer?,
+                      UnsafeMutableRawPointer?,
+                      UnsafeMutableRawPointer?) -> CInt
+}
```

Comment:
What's the cycle here? Just curious.

Reply:
DeclContext::getGenericSignature() => ... => TypeAliasDecl::getStructuralType() => TypeChecker::diagnoseInvalidFunctionType() => TypeBase::isRepresentable() => DeclContext::getGenericSignature()

I could fix isRepresentable() but there's no reason to perform semantic checks in structural mode at all, since we're going to revisit it in interface type mode anyway.